crates/sui-json-rpc/src/read_api.rs

```diff
@@ -239,48 +230,21 @@ impl SuiRpcModule for ReadApi {
 #[async_trait]
 impl RpcFullNodeReadApiServer for FullNodeApi {
     async fn dev_inspect_transaction(
-        &self,
-        tx_bytes: Base64,
-        epoch: Option<EpochId>,
-    ) -> RpcResult<DevInspectResults> {
-        let epoch = match epoch {
-            None => self.get_sui_system_state_object_epoch()?,
-            Some(n) => n,
-        };
-        let (txn_data, txn_digest) = get_transaction_data_and_digest(tx_bytes)?;
-        Ok(self
-            .state
-            .dev_inspect_transaction(txn_data, txn_digest, epoch)
-            .await?)
-    }
-
-    async fn dev_inspect_move_call(
         &self,
         sender_address: SuiAddress,
-        package_object_id: ObjectID,
-        module: String,
-        function: String,
-        type_arguments: Vec<SuiTypeTag>,
-        arguments: Vec<SuiJsonValue>,
+        tx_bytes: Base64,
+        gas_price: Option<u64>,
         epoch: Option<EpochId>,
     ) -> RpcResult<DevInspectResults> {
         let epoch = match epoch {
             None => self.get_sui_system_state_object_epoch()?,
             Some(n) => n,
         };
-        let move_call = self
-            .dev_inspect_builder
-            .single_move_call(
-                package_object_id,
-                &module,
-                &function,
-                type_arguments,
-                arguments,
-            )
-            .await?;
+        let tx_kind: TransactionKind =
+            bcs::from_bytes(&tx_bytes.to_vec().map_err(|e| anyhow!(e))?).map_err(|e| anyhow!(e))?;
         Ok(self
             .state
-            .dev_inspect_move_call(sender_address, move_call, epoch)
+            .dev_inspect_transaction(sender_address, tx_kind, gas_price.unwrap_or(1), epoch)
```

Comment:
```suggestion
            .dev_inspect_transaction(sender_address, tx_kind, gas_price.unwrap_or(1), epoch)
```

we should use reference gas price here as the default. I'll make this change in my diff as well unless there's any concern